aiopg/transaction.py

```diff
@@ -77,10 +94,26 @@ def begin(self):
         return query
 
 
+class DefaultCompiler(IsolationCompiler):
+    __slots__ = ()
+
+    def __init__(self, readonly, deferrable):
+        if readonly or deferrable:
+            raise ValueError("Readonly or deferrable are not supported")
```

Comment:
Why readonly and deferrable are not supported? As this object is running `BEGIN` it should be able to support it, they are independent from the isolation level. 

https://www.postgresql.org/docs/current/sql-begin.html

Reply:
I think it was right that `__slots__` was on the base class and that these properties are available to all the hierarchy

Reply:
![image](https://user-images.githubusercontent.com/664889/102718189-a6abac00-4308-11eb-9e1b-17fcf4155074.png)

I've only changed the current implementation 🙈

Reply:
>Why readonly and deferrable are not supported? As this object is running BEGIN it should be able to support it, they are independent from the isolation level.

I dunno why it was unsupported, but let me check

Reply:
![image](https://user-images.githubusercontent.com/664889/102718413-ca232680-4309-11eb-917a-a5d96abfbd08.png)

https://www.postgresql.org/docs/13/sql-set-transaction.html

It looks like readonly supported for any isolation level, but deferred - only for serialisable with readonly.

But I'm a but unsure about current behaviour with raise of an exception.

Reply:
You shouldn't impose a rule on top of what postgres implements.

If it's a no-op it may be a no-op only for the current implementation and change in the future. If the database accepts it, it is not the driver's business to decide to raise an exception.

This seems to work: the database doesn't raise exceptions or even warnings:

```
piro=# begin isolation level read uncommitted deferrable;
BEGIN
*piro=#
```

so I don't see a reason to special-case it and raise an exception.

Reply:
Agree, thanks 🎉